libespm/src/CombineBrowser.cpp

```diff
@@ -84,6 +84,34 @@ CombineBrowser::GetRecordsByType(const char* type) const
   return res;
 }
 
+const std::vector<const RecordHeader*>
+CombineBrowser::GetDistinctRecordsByType(const char* type) const
+{
+  const auto allRecordsSrc = GetRecordsByType(type);
+  if (allRecordsSrc.size() == 0) {
+    return {};
+  }
+
+  size_t approxSize = 0;
+  for (const auto recSrc : allRecordsSrc) {
+    approxSize = std::max(approxSize, recSrc->size());
+  }
+
+  spp::sparse_hash_set<formId> formSet(approxSize);
+  std::vector<const RecordHeader*> result;
+  result.reserve(approxSize);
+
+  for (auto i = allRecordsSrc.size() - 1; i != static_cast<size_t>(-1); --i) {
+    for (auto rec : *allRecordsSrc[i]) {
+      if (formSet.insert(rec->GetId()).second) {
```

Comment:
looks incorrect due to direct GetId() usage. GetId() doesn't respect global load order. it's relative to its file's load order.

it doesn't make sense to make a set from form ids from different files

should be converted into global form id